commands/account/auth.ts

```diff
@@ -0,0 +1,183 @@
+import { Argv, ArgumentsCamelCase } from 'yargs';
+import {
+  loadConfig,
+  getConfigPath,
+  configFileExists,
+  updateAccountConfig,
+  writeConfig,
+  createEmptyConfigFile,
+  deleteEmptyConfigFile,
+  getConfigDefaultAccount,
+} from '@hubspot/local-dev-lib/config';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import {
+  getAccessToken,
+  updateConfigWithAccessToken,
+} from '@hubspot/local-dev-lib/personalAccessKey';
+import { ENVIRONMENTS } from '@hubspot/local-dev-lib/constants/environments';
+import { toKebabCase } from '@hubspot/local-dev-lib/text';
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import { CLIAccount_NEW } from '@hubspot/local-dev-lib/types/Accounts';
+
+import { addConfigOptions, addGlobalOptions } from '../../lib/commonOpts';
+import { handleExit } from '../../lib/process';
+import { logError } from '../../lib/errorHandlers/index';
+import { i18n } from '../../lib/lang';
+import { trackCommandUsage } from '../../lib/usageTracking';
+import { addTestingOptions } from '../../lib/commonOpts';
+import { personalAccessKeyPrompt } from '../../lib/prompts/personalAccessKeyPrompt';
+import { cliAccountNamePrompt } from '../../lib/prompts/accountNamePrompt';
+import { setAsDefaultAccountPrompt } from '../../lib/prompts/setAsDefaultAccountPrompt';
+import { EXIT_CODES } from '../../lib/enums/exitCodes';
+import { uiFeatureHighlight } from '../../lib/ui';
+import { CommonArgs, ConfigArgs } from '../../types/Yargs';
+
+const i18nKey = 'commands.account.subcommands.auth';
+
+async function createOrUpdateConfig(
+  env: Environment,
+  doesConfigExist: boolean,
+  account?: number
+): Promise<CLIAccount_NEW | null> {
+  try {
+    const { personalAccessKey } = await personalAccessKeyPrompt({
+      env,
+      account,
+    });
+    const token = await getAccessToken(personalAccessKey, env);
+    const defaultName = token.hubName ? toKebabCase(token.hubName) : null;
+
+    const name = doesConfigExist
+      ? undefined
+      : (await cliAccountNamePrompt(defaultName)).name;
+
+    const updatedConfig = await updateConfigWithAccessToken(
+      token,
+      personalAccessKey,
+      env,
+      name,
+      !doesConfigExist
+    );
+
+    if (!updatedConfig) return null;
+
+    if (doesConfigExist && !updatedConfig.name) {
```

Comment:
Is this line setting a name on an account if it already existed in the config, but didn't have a name yet?

Reply:
Yes. Basically, there is a complicated set of scenarios when users update the config file:

1) If the user is creating the config for the first time, we want to prompt them for an account name first thing.

2) If the user is re-authorizing an account that is already in the config, we do not want to prompt them for the account name.

3) If the user is authorizing an account that is NOT already in the config, we do want to prompt them for the account name. 

This conditional on line 64 covers scenarios 2 and 3. I'll see whether I can't simplify this and combine the conditional on line 68 with this one.

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1365/commits/2acdfb8eb4d268b18e4e76619c9101f2aa19d8bc